src/sage/algebras/lie_algebras/quotient.py

```diff
@@ -218,8 +218,11 @@ def __classcall_private__(cls, I, ambient=None, names=None,
         index_set = [i for i in sorted_indices if i not in I_supp]
 
         if names is None:
-            amb_names = dict(zip(sorted_indices, ambient.variable_names()))
-            names = [amb_names[i] for i in index_set]
+            try:
+                amb_names = dict(zip(sorted_indices, ambient.variable_names()))
+                names = [amb_names[i] for i in index_set]
+            except ValueError:  # ambient has not assigned variable names
+                pass
```

Comment:
I'd suggest to rewrite it as `try... except... else`, which makes it clearer which of the lines is expected to raise the exception

Reply:
I instead just examined the `_names` attribute and avoided using `try`-`except` to control programming flow.